crypto/ec/ecdsa_ossl.c

```diff
@@ -126,22 +127,33 @@ static int ecdsa_sign_setup(EC_KEY *eckey, BN_CTX *ctx_in,
         goto err;
 
     do {
-        /* get random k */
+        /* get random or deterministic k */
         do {
             if (dgst != NULL) {
-                if (!BN_generate_dsa_nonce(k, order, priv_key,
-                                           dgst, dlen, ctx)) {
-                    ECerr(EC_F_ECDSA_SIGN_SETUP,
-                          EC_R_RANDOM_NUMBER_GENERATION_FAILED);
-                    goto err;
+                if(hashnid <= 0) { /* random */
+                    kgen = BN_generate_dsa_nonce(k, order, priv_key,
+                                                 dgst, dlen, ctx);
+                } else { /* deterministic */
+#ifndef FIPS_MODE
+                    kgen = bn_generate_dsa_deterministic_nonce(k, order, priv_key,
+                                                               dgst, dlen, hashnid, ctx);
+#else
+                    kgen = 0; /* FIPS_MODE is not compatiable with deterministic ECDSA signature */
```

Comment:
why not just set an error and break?

Reply:
It is just doing what you said...
It set kgen=0 and ECErr here, it will break and goto err label in line 153-155